mimic/model/mailgun_objects.py

```diff
@@ -0,0 +1,119 @@
+"""
+Mailgun object storage
+"""
+import time
+from characteristic import attributes, Attribute
+
+
+@attributes(["message_id", "to", "msg_from", "subject", "body",
+             Attribute("headers", default_factory=dict)])
+class Messages(object):
+    """
+    A :obj:`Message` is a representation of an email in Mailgun.
+    It can produce JSON-serializable objects for various pieces of
+    state that are required for API responses.
+    """
+
+    static_defaults = {
+        "tags": [],
+        "delivery-status": {
+            "message": "",
+            "code": 0,
+            "description": None,
+            "session-seconds": 1.114408016204834
+        },
+        "envelope": {
+            "transport": "smtp",
+            "sending-ip": "127.0.0.1",
+        },
+        "recipient-domain": "mailgun.com",
+        "id": "mimic-LCZuENBlS0iWjs-yBpNJaQ",
+        "campaigns": [],
+        "user-variables": {},
+        "flags": {
+            "is-routed": None,
+            "is-authenticated": True,
+            "is-system-test": False,
+            "is-test-mode": False
+        },
+        "log-level": "info",
+        "timestamp": time.time(),
+        "message": {
+            "headers": {},
+            "attachments": [],
+            "recipients": [],
+            "size": 0
+        },
+        "recipient": None,
+        "event": "delivered"
+    }
+
+    def generate_events(self):
+        """
+        Long-form JSON-serializable object representation of this message, as
+        returned by a GET on this individual message.
+        """
+        template = self.static_defaults.copy()
+        template.update({
+            "envelope": {
+                "sender": self.msg_from,
+                "targets": self.to
+            },
+            "message": {
+                "headers": {
+                    "to": self.to,
+                    "message-id": self.message_id,
+                    "from": self.msg_from,
+                    "subject": self.subject
+                },
+                "recipients": [self.to],
+                "recipient": self.to
+            }
+        })
+        return template
+
+
+@attributes([Attribute("message_store", default_factory=list)])
+class MessageStore(object):
+    """
+    A collection of message objects.
+    """
+
+    def _add_to_message_store(self, **attributes):
```

Comment:
Non-blocking question - out of curiosity, why a private function if it's called by the REST api?